packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -1045,14 +1039,10 @@ export function diffHydratedProperties(
       if (suppressHydrationWarning) {
         // Don't bother comparing. We're ignoring all these warnings.
       } else if (
-        (enableFlareAPI && propKey === LISTENERS) ||
-        propKey === SUPPRESS_CONTENT_EDITABLE_WARNING ||
-        propKey === SUPPRESS_HYDRATION_WARNING ||
+        enableFlareAPIIsSet(propKey) ||
         // Controlled attributes are not validated
         // TODO: Only ignore them on controlled tags.
-        propKey === 'value' ||
-        propKey === 'checked' ||
-        propKey === 'selected'
+        ['value', 'checked', 'selected'].includes(propKey)
```

Comment:
This is far less efficient than using simple or conditions.

Reply:
should I revert?

Reply:
Yep